### ISSUE TITLE
Grafana-server vulnerability

### ISSUE THREAD

**Issue by BrianEdmund** [hashcloak/Meson#150]
Running a Snyk scan on the grafana image results with a high severity vulnerability found in github.com/ua-parser/uap-go/uaparser caused by Regular Expression Denial of Service (ReDoS)

**Comment by sc0Vu** [comment]
Hi @BrianEdmund 

Thanks, after I rescan the grafana:8.4.3 docker image we used, it turns out the uaparser is medium level (yaml is high level).

```
Testing grafana/grafana:8.4.3...

✗ Medium severity vulnerability found in github.com/ua-parser/uap-go/uaparser
  Description: Regular Expression Denial of Service (ReDoS)
  Info: https://snyk.io/vuln/SNYK-GOLANG-GITHUBCOMUAPARSERUAPGOUAPARSER-1298048
  Introduced through: github.com/ua-parser/uap-go/uaparser@#00c877edfe0f
  From: github.com/ua-parser/uap-go/uaparser@#00c877edfe0f

✗ High severity vulnerability found in gopkg.in/yaml.v3
  Description: Denial of Service (DoS)
  Info: https://snyk.io/vuln/SNYK-GOLANG-GOPKGINYAMLV3-2841557
  Introduced through: gopkg.in/yaml.v3@#496545a6307b
  From: gopkg.in/yaml.v3@#496545a6307b
  Fixed in: 3.0.0
```

For the high level severity vulnerability, I think they already fixed. For the medium level issue (https://security.snyk.io/vuln/SNYK-GOLANG-GITHUBCOMUAPARSERUAPGOUAPARSER-1298048), they only use uaparser in `GET /user/auth-tokens` (https://github.com/grafana/grafana/blob/a14621fff679add99c494628f1fb9a001a734f47/pkg/api/user_token.go#L29), they said it was fixed in the master branch and grafana did use their latest commit (00c877edfe0f). I'm not really sure about whether this bug is still there.

Anyway, I'll update grafana images and check other images.

**Comment by sc0Vu** [comment]
Found some minor vulnerability in meson server and katzenmint pki:

```
## meson/server 

Testing meson/server...

✗ Critical severity vulnerability found in zlib/zlib
  Description: Out-of-bounds Write
  Info: https://snyk.io/vuln/SNYK-ALPINE316-ZLIB-2976176
  Introduced through: zlib/zlib@1.2.12-r1, apk-tools/apk-tools@2.12.9-r3
  From: zlib/zlib@1.2.12-r1
  From: apk-tools/apk-tools@2.12.9-r3 > zlib/zlib@1.2.12-r1
  Image layer: Introduced by your base image (alpine)
  Fixed in: 1.2.12-r2



Package manager:   apk
Target file:       Dockerfile.server
Project name:      docker-image|meson/server
Docker image:      meson/server
Platform:          linux/arm64
Base image:        alpine

Tested 16 dependencies for known vulnerabilities, found 1 vulnerability.

Your base image is out of date
1) Pull the latest version of your base image by running 'docker pull alpine:latest'
2) Rebuild your local image

For more free scans that keep your images secure, sign up to Snyk at https://dockr.ly/3ePqVcp


## katzenmint/pki

Testing katzenmint/pki...

✗ Low severity vulnerability found in curl/libcurl
  Description: CVE-2022-35252
  Info: https://snyk.io/vuln/SNYK-ALPINE316-CURL-3011748
  Introduced through: curl/libcurl@7.83.1-r2, curl/curl@7.83.1-r2
  From: curl/libcurl@7.83.1-r2
  From: curl/curl@7.83.1-r2 > curl/libcurl@7.83.1-r2
  From: curl/curl@7.83.1-r2
  Image layer: 'apk add --no-cache ca-certificates tzdata curl'
  Fixed in: 7.83.1-r3

✗ Critical severity vulnerability found in zlib/zlib
  Description: Out-of-bounds Write
  Info: https://snyk.io/vuln/SNYK-ALPINE316-ZLIB-2976176
  Introduced through: zlib/zlib@1.2.12-r1, apk-tools/apk-tools@2.12.9-r3, curl/libcurl@7.83.1-r2, curl/curl@7.83.1-r2
  From: zlib/zlib@1.2.12-r1
  From: apk-tools/apk-tools@2.12.9-r3 > zlib/zlib@1.2.12-r1
  From: curl/libcurl@7.83.1-r2 > zlib/zlib@1.2.12-r1
  and 1 more...
  Image layer: 'apk add --no-cache ca-certificates tzdata curl'
  Fixed in: 1.2.12-r2



Package manager:   apk
Target file:       Dockerfile.katzenmint
Project name:      docker-image|katzenmint/pki
Docker image:      katzenmint/pki
Platform:          linux/arm64
Base image:        alpine

Tested 20 dependencies for known vulnerabilities, found 2 vulnerabilities.

Your base image is out of date
1) Pull the latest version of your base image by running 'docker pull alpine:latest'
2) Rebuild your local image

For more free scans that keep your images secure, sign up to Snyk at https://dockr.ly/3ePqVcp
```

**Comment by sc0Vu** [comment]
The issue was for apk tools, this should be fine. Also it looks great after upgrade the version of alpine.

Before rebuild.
```BASH
Testing katzenmint/pki...

✗ Low severity vulnerability found in openssl/libcrypto3
  Description: CVE-2023-0401
  Info: https://snyk.io/vuln/SNYK-ALPINE317-OPENSSL-3314647
  Introduced through: openssl/libcrypto3@3.0.7-r2, openssl/libssl3@3.0.7-r2, apk-tools/apk-tools@2.12.10-r1, busybox/ssl_client@1.35.0-r29, ca-certificates/ca-certificates@20220614-r4, curl/libcurl@7.87.0-r1
  From: openssl/libcrypto3@3.0.7-r2
  From: openssl/libssl3@3.0.7-r2 > openssl/libcrypto3@3.0.7-r2
  From: apk-tools/apk-tools@2.12.10-r1 > openssl/libcrypto3@3.0.7-r2
  and 7 more...
  Image layer: 'apk add --no-cache ca-certificates tzdata curl'
  Fixed in: 3.0.8-r0

✗ Low severity vulnerability found in openssl/libcrypto3
  Description: CVE-2022-4203
  Info: https://snyk.io/vuln/SNYK-ALPINE317-OPENSSL-3314648
  Introduced through: openssl/libcrypto3@3.0.7-r2, openssl/libssl3@3.0.7-r2, apk-tools/apk-tools@2.12.10-r1, busybox/ssl_client@1.35.0-r29, ca-certificates/ca-certificates@20220614-r4, curl/libcurl@7.87.0-r1
  From: openssl/libcrypto3@3.0.7-r2
  From: openssl/libssl3@3.0.7-r2 > openssl/libcrypto3@3.0.7-r2
  From: apk-tools/apk-tools@2.12.10-r1 > openssl/libcrypto3@3.0.7-r2
  and 7 more...
  Image layer: 'apk add --no-cache ca-certificates tzdata curl'
  Fixed in: 3.0.8-r0

✗ Low severity vulnerability found in openssl/libcrypto3
  Description: CVE-2023-0215
  Info: https://snyk.io/vuln/SNYK-ALPINE317-OPENSSL-3314650
  Introduced through: openssl/libcrypto3@3.0.7-r2, openssl/libssl3@3.0.7-r2, apk-tools/apk-tools@2.12.10-r1, busybox/ssl_client@1.35.0-r29, ca-certificates/ca-certificates@20220614-r4, curl/libcurl@7.87.0-r1
  From: openssl/libcrypto3@3.0.7-r2
  From: openssl/libssl3@3.0.7-r2 > openssl/libcrypto3@3.0.7-r2
  From: apk-tools/apk-tools@2.12.10-r1 > openssl/libcrypto3@3.0.7-r2
  and 7 more...
  Image layer: 'apk add --no-cache ca-certificates tzdata curl'
  Fixed in: 3.0.8-r0

✗ Low severity vulnerability found in openssl/libcrypto3
  Description: CVE-2023-0286
  Info: https://snyk.io/vuln/SNYK-ALPINE317-OPENSSL-3314651
  Introduced through: openssl/libcrypto3@3.0.7-r2, openssl/libssl3@3.0.7-r2, apk-tools/apk-tools@2.12.10-r1, busybox/ssl_client@1.35.0-r29, ca-certificates/ca-certificates@20220614-r4, curl/libcurl@7.87.0-r1
  From: openssl/libcrypto3@3.0.7-r2
  From: openssl/libssl3@3.0.7-r2 > openssl/libcrypto3@3.0.7-r2
  From: apk-tools/apk-tools@2.12.10-r1 > openssl/libcrypto3@3.0.7-r2
  and 7 more...
  Image layer: 'apk add --no-cache ca-certificates tzdata curl'
  Fixed in: 3.0.8-r0

✗ Low severity vulnerability found in openssl/libcrypto3
  Description: CVE-2022-4450
  Info: https://snyk.io/vuln/SNYK-ALPINE317-OPENSSL-3314657
  Introduced through: openssl/libcrypto3@3.0.7-r2, openssl/libssl3@3.0.7-r2, apk-tools/apk-tools@2.12.10-r1, busybox/ssl_client@1.35.0-r29, ca-certificates/ca-certificates@20220614-r4, curl/libcurl@7.87.0-r1
  From: openssl/libcrypto3@3.0.7-r2
  From: openssl/libssl3@3.0.7-r2 > openssl/libcrypto3@3.0.7-r2
  From: apk-tools/apk-tools@2.12.10-r1 > openssl/libcrypto3@3.0.7-r2
  and 7 more...
  Image layer: 'apk add --no-cache ca-certificates tzdata curl'
  Fixed in: 3.0.8-r0

✗ Low severity vulnerability found in openssl/libcrypto3
  Description: CVE-2023-0216
  Info: https://snyk.io/vuln/SNYK-ALPINE317-OPENSSL-3314658
  Introduced through: openssl/libcrypto3@3.0.7-r2, openssl/libssl3@3.0.7-r2, apk-tools/apk-tools@2.12.10-r1, busybox/ssl_client@1.35.0-r29, ca-certificates/ca-certificates@20220614-r4, curl/libcurl@7.87.0-r1
  From: openssl/libcrypto3@3.0.7-r2
  From: openssl/libssl3@3.0.7-r2 > openssl/libcrypto3@3.0.7-r2
  From: apk-tools/apk-tools@2.12.10-r1 > openssl/libcrypto3@3.0.7-r2
  and 7 more...
  Image layer: 'apk add --no-cache ca-certificates tzdata curl'
  Fixed in: 3.0.8-r0

✗ Low severity vulnerability found in openssl/libcrypto3
  Description: CVE-2022-4304
  Info: https://snyk.io/vuln/SNYK-ALPINE317-OPENSSL-3314659
  Introduced through: openssl/libcrypto3@3.0.7-r2, openssl/libssl3@3.0.7-r2, apk-tools/apk-tools@2.12.10-r1, busybox/ssl_client@1.35.0-r29, ca-certificates/ca-certificates@20220614-r4, curl/libcurl@7.87.0-r1
  From: openssl/libcrypto3@3.0.7-r2
  From: openssl/libssl3@3.0.7-r2 > openssl/libcrypto3@3.0.7-r2
  From: apk-tools/apk-tools@2.12.10-r1 > openssl/libcrypto3@3.0.7-r2
  and 7 more...
  Image layer: 'apk add --no-cache ca-certificates tzdata curl'
  Fixed in: 3.0.8-r0

✗ Low severity vulnerability found in openssl/libcrypto3
  Description: CVE-2023-0217
  Info: https://snyk.io/vuln/SNYK-ALPINE317-OPENSSL-3314660
  Introduced through: openssl/libcrypto3@3.0.7-r2, openssl/libssl3@3.0.7-r2, apk-tools/apk-tools@2.12.10-r1, busybox/ssl_client@1.35.0-r29, ca-certificates/ca-certificates@20220614-r4, curl/libcurl@7.87.0-r1
  From: openssl/libcrypto3@3.0.7-r2
  From: openssl/libssl3@3.0.7-r2 > openssl/libcrypto3@3.0.7-r2
  From: apk-tools/apk-tools@2.12.10-r1 > openssl/libcrypto3@3.0.7-r2
  and 7 more...
  Image layer: 'apk add --no-cache ca-certificates tzdata curl'
  Fixed in: 3.0.8-r0

Testing meson/server...

✗ Low severity vulnerability found in openssl/libcrypto3
  Description: CVE-2023-0401
  Info: https://snyk.io/vuln/SNYK-ALPINE317-OPENSSL-3314647
  Introduced through: openssl/libcrypto3@3.0.7-r2, openssl/libssl3@3.0.7-r2, apk-tools/apk-tools@2.12.10-r1, busybox/ssl_client@1.35.0-r29, ca-certificates/ca-certificates@20220614-r4
  From: openssl/libcrypto3@3.0.7-r2
  From: openssl/libssl3@3.0.7-r2 > openssl/libcrypto3@3.0.7-r2
  From: apk-tools/apk-tools@2.12.10-r1 > openssl/libcrypto3@3.0.7-r2
  and 5 more...
  Image layer: 'apk add --no-cache ca-certificates tzdata'
  Fixed in: 3.0.8-r0

✗ Low severity vulnerability found in openssl/libcrypto3
  Description: CVE-2022-4203
  Info: https://snyk.io/vuln/SNYK-ALPINE317-OPENSSL-3314648
  Introduced through: openssl/libcrypto3@3.0.7-r2, openssl/libssl3@3.0.7-r2, apk-tools/apk-tools@2.12.10-r1, busybox/ssl_client@1.35.0-r29, ca-certificates/ca-certificates@20220614-r4
  From: openssl/libcrypto3@3.0.7-r2
  From: openssl/libssl3@3.0.7-r2 > openssl/libcrypto3@3.0.7-r2
  From: apk-tools/apk-tools@2.12.10-r1 > openssl/libcrypto3@3.0.7-r2
  and 5 more...
  Image layer: 'apk add --no-cache ca-certificates tzdata'
  Fixed in: 3.0.8-r0

✗ Low severity vulnerability found in openssl/libcrypto3
  Description: CVE-2023-0215
  Info: https://snyk.io/vuln/SNYK-ALPINE317-OPENSSL-3314650
  Introduced through: openssl/libcrypto3@3.0.7-r2, openssl/libssl3@3.0.7-r2, apk-tools/apk-tools@2.12.10-r1, busybox/ssl_client@1.35.0-r29, ca-certificates/ca-certificates@20220614-r4
  From: openssl/libcrypto3@3.0.7-r2
  From: openssl/libssl3@3.0.7-r2 > openssl/libcrypto3@3.0.7-r2
  From: apk-tools/apk-tools@2.12.10-r1 > openssl/libcrypto3@3.0.7-r2
  and 5 more...
  Image layer: 'apk add --no-cache ca-certificates tzdata'
  Fixed in: 3.0.8-r0

✗ Low severity vulnerability found in openssl/libcrypto3
  Description: CVE-2023-0286
  Info: https://snyk.io/vuln/SNYK-ALPINE317-OPENSSL-3314651
  Introduced through: openssl/libcrypto3@3.0.7-r2, openssl/libssl3@3.0.7-r2, apk-tools/apk-tools@2.12.10-r1, busybox/ssl_client@1.35.0-r29, ca-certificates/ca-certificates@20220614-r4
  From: openssl/libcrypto3@3.0.7-r2
  From: openssl/libssl3@3.0.7-r2 > openssl/libcrypto3@3.0.7-r2
  From: apk-tools/apk-tools@2.12.10-r1 > openssl/libcrypto3@3.0.7-r2
  and 5 more...
  Image layer: 'apk add --no-cache ca-certificates tzdata'
  Fixed in: 3.0.8-r0

✗ Low severity vulnerability found in openssl/libcrypto3
  Description: CVE-2022-4450
  Info: https://snyk.io/vuln/SNYK-ALPINE317-OPENSSL-3314657
  Introduced through: openssl/libcrypto3@3.0.7-r2, openssl/libssl3@3.0.7-r2, apk-tools/apk-tools@2.12.10-r1, busybox/ssl_client@1.35.0-r29, ca-certificates/ca-certificates@20220614-r4
  From: openssl/libcrypto3@3.0.7-r2
  From: openssl/libssl3@3.0.7-r2 > openssl/libcrypto3@3.0.7-r2
  From: apk-tools/apk-tools@2.12.10-r1 > openssl/libcrypto3@3.0.7-r2
  and 5 more...
  Image layer: 'apk add --no-cache ca-certificates tzdata'
  Fixed in: 3.0.8-r0

✗ Low severity vulnerability found in openssl/libcrypto3
  Description: CVE-2023-0216
  Info: https://snyk.io/vuln/SNYK-ALPINE317-OPENSSL-3314658
  Introduced through: openssl/libcrypto3@3.0.7-r2, openssl/libssl3@3.0.7-r2, apk-tools/apk-tools@2.12.10-r1, busybox/ssl_client@1.35.0-r29, ca-certificates/ca-certificates@20220614-r4
  From: openssl/libcrypto3@3.0.7-r2
  From: openssl/libssl3@3.0.7-r2 > openssl/libcrypto3@3.0.7-r2
  From: apk-tools/apk-tools@2.12.10-r1 > openssl/libcrypto3@3.0.7-r2
  and 5 more...
  Image layer: 'apk add --no-cache ca-certificates tzdata'
  Fixed in: 3.0.8-r0

✗ Low severity vulnerability found in openssl/libcrypto3
  Description: CVE-2022-4304
  Info: https://snyk.io/vuln/SNYK-ALPINE317-OPENSSL-3314659
  Introduced through: openssl/libcrypto3@3.0.7-r2, openssl/libssl3@3.0.7-r2, apk-tools/apk-tools@2.12.10-r1, busybox/ssl_client@1.35.0-r29, ca-certificates/ca-certificates@20220614-r4
  From: openssl/libcrypto3@3.0.7-r2
  From: openssl/libssl3@3.0.7-r2 > openssl/libcrypto3@3.0.7-r2
  From: apk-tools/apk-tools@2.12.10-r1 > openssl/libcrypto3@3.0.7-r2
  and 5 more...
  Image layer: 'apk add --no-cache ca-certificates tzdata'
  Fixed in: 3.0.8-r0

✗ Low severity vulnerability found in openssl/libcrypto3
  Description: CVE-2023-0217
  Info: https://snyk.io/vuln/SNYK-ALPINE317-OPENSSL-3314660
  Introduced through: openssl/libcrypto3@3.0.7-r2, openssl/libssl3@3.0.7-r2, apk-tools/apk-tools@2.12.10-r1, busybox/ssl_client@1.35.0-r29, ca-certificates/ca-certificates@20220614-r4
  From: openssl/libcrypto3@3.0.7-r2
  From: openssl/libssl3@3.0.7-r2 > openssl/libcrypto3@3.0.7-r2
  From: apk-tools/apk-tools@2.12.10-r1 > openssl/libcrypto3@3.0.7-r2
  and 5 more...
  Image layer: 'apk add --no-cache ca-certificates tzdata'
  Fixed in: 3.0.8-r0
```

After rebuild.
```BASH
Testing katzenmint/pki...

Organization:      cclemonpie
Package manager:   apk
Project name:      docker-image|katzenmint/pki
Docker image:      katzenmint/pki
Platform:          linux/arm64
Licenses:          enabled

✔ Tested 21 dependencies for known issues, no vulnerable paths found.

Testing meson/server...

Organization:      cclemonpie
Package manager:   apk
Project name:      docker-image|meson/server
Docker image:      meson/server
Platform:          linux/arm64
Licenses:          enabled

✔ Tested 17 dependencies for known issues, no vulnerable paths found.
```
